### PR TITLE
Add fsck keymaps for eco, planck and preonic

### PR DESCRIPTION
**Pull request by fsck**
## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by this PR

* Add fsck keymaps for the eco, planck and preonic keyboards

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/eco/keymaps/fsck/keymap.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#define JC_C    GUI_T(KC_C)
+#define JC_COMM GUI_T(KC_COMM)
+#define JC_DOT  ALT_T(KC_DOT)
+#define JC_SLSH CTL_T(KC_SLSH)
+
+#define JC_CAPS LT(_RAISE,KC_CAPS)
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
+#define XXXXXXX KC_NO
```
Comment (fauxpark) [inline keyboards/eco/keymaps/fsck/keymap.c]: These are already defined in QMK, you don't need to have them here :)

File: keyboards/planck/keymaps/fsck/keymap.c [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
```
Comment (fauxpark) [inline keyboards/planck/keymaps/fsck/keymap.c]: `break` after `return` is pointless, you can remove these.

File: keyboards/planck/keymaps/fsck/keymap.c [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+ * | Tab  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Bksp |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Esc  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Ctrl |Brite | GUI  | Alt  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = {
```
Comment (drashna) [inline keyboards/planck/keymaps/fsck/keymap.c]: Could you update this to use the `LAYOUT` format, as in the default keymap?

File: keyboards/planck/keymaps/fsck/rules.mk [hunk 4]
```diff
@@ -0,0 +1,5 @@
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/planck/keymaps/fsck/rules.mk]: Could you remove this (ifndef/include/endif) block? 

It's not needed, and is a holdover from the old build system.

File: keyboards/preonic/keymaps/fsck/config.h [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/preonic/keymaps/fsck/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/preonic/keymaps/fsck/config.h [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+#define CONFIG_USER_H
```
Comment (drashna) [inline keyboards/preonic/keymaps/fsck/config.h]:
```suggestion
```

File: keyboards/preonic/keymaps/fsck/config.h [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/preonic/keymaps/fsck/config.h]:
```suggestion
```

File: keyboards/preonic/keymaps/fsck/config.h [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
+#endif
```
Comment (drashna) [inline keyboards/preonic/keymaps/fsck/config.h]:
```suggestion
```

**Comment by fsck** [comment]
Thank you both for your detailed feedback, I have updated the pr with your suggested changes.

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**